### PR TITLE
Set NIX_PATH in shell.nix

### PR DESCRIPTION
**Pull request by r2r-dev**
This PR simplifies `flake.nix` a bit, ensures that our overlays are available in entrypoint shell, and moves `NIX_PATH` set-up from `shell.sh` to `shell.nix`.

### CODE REVIEW

File: scripts/shell.nix [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+      let lock = fromJSON (readFile ../flake.lock); in
+      fetchTarball {
+        url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
+        sha256 = lock.nodes.flake-compat.locked.narHash;
+      }
+    )
+    { src = ../.; }
+  ).defaultNix;
+
+  NIX_PATH =
```
Comment (borancar) [inline scripts/shell.nix:15]: I like this!

File: flake.nix [hunk 2]
```diff
@@ -18,28 +18,14 @@
...
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ nix-direnv.overlay ];
+        callArgs = { inherit system overlays; };
+      in
+      { inherit callArgs; } //
+      builtins.mapAttrs
+        (n: v: import v callArgs)
+        { inherit nixpkgs nixpkgs_latest; }
+    );
```
Comment (borancar) [inline flake.nix:30]: We no longer need devShell, right?

**Review by borancar — APPROVED** [review]

File: flake.nix (hunk 2 above)
Comment (r2r-dev) [reply flake.nix:30]: Yeah. I don't think it makes sense to keep it. Especially that, we explicitly call `shell.nix` from `nix-shell.sh`.

**Review by AleksanderGondek — APPROVED** [review]
Awesome, love the changes!

**Pull request merged**